src/util/shm/mpir_shm_coll.c

```diff
@@ -3,36 +3,156 @@
  *  (C) 2019 by Argonne National Laboratory.
  *      See COPYRIGHT in top-level directory.
  */
+#include <mpidimpl.h>
 #include "mpir_shm.h"
 #include "mpir_shm_coll.h"
+#include "mpir_mem.h"
+#include "build_nodemap.h"
+
+#define PROC_MEM_LEN (MPIR_SHM_CACHE_LINE_LEN * 16)
```

Comment:
@gcongiu What is the significance of  `16` here?

Reply:
@jain-surabhi-23 this is just to allocate enough space for every process in the shm segment (`1KB` currently). This should be enough for all cases as at init time, as far as I know, we only need to communicate single elements (mostly integers).